File: PointOfSalesV2/PointOfSalesV2.Entities/Product.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using System.ComponentModel.DataAnnotations.Schema;

namespace PointOfSalesV2.Entities
{
    public class Product : CommonData
    {


        public string Description { get; set; }
        public decimal Price { get; set; }
        public decimal Price2 { get; set; }
        public decimal Price3 { get; set; }

        public string Code { get; set; }
        public string DetailsClass { get; set; }

        public bool IsService { get; set; }
        public bool IsCompositeProduct { get; set; }

        public decimal Cost { get; set; }
        public decimal SellerRate { get; set; } = 0;

        public long CurrencyId { get; set; }
        public decimal Existence { get; set; }
      
        public string Name { get; set; }

      

        public string Details { get; set; }





        [ForeignKey("CurrencyId")]
        public Currency Currency { get; set; }

      

     
        public List<ProductTax> Taxes { get; set; }





      
        public List<UnitProductEquivalence> ProductUnits { get; set; }

   
        public List<CompositeProduct> BaseCompositeProducts { get; set; }



    }
}

File: PointOfSalesV2/PointOfSalesV2.FrontEnd/ClientApp/src/app/@core/data/product.ts
import { BaseModel } from './baseModel';
import { Currency } from './currencyModel';
import { Tax } from './taxModel';
import { Unit } from './unitModel';


export class Product extends BaseModel {

    description:string;
    price:number;
    price2:number;
    price3:number;
    code:string;
    isService:boolean;
    isCompositeProduct:boolean;
    cost:number;
    sellerRate:number;
    currencyId:number;
    existence:number;
    name:string;
    details:string;
    currency :Currency
    taxes:any[];
    productUnits:any[];
    baseCompositeProducts:any[];
    suppliersCosts:any[]=[];
}
export class ProductTax extends BaseModel {

    productId:number;
    taxId:number;
    tax:Tax;
}

export class UnitProductEquivalence extends BaseModel {

         unitId:number; 
         productId:number; 
         equivalence:number;
         isPrimary :boolean;
         order :number; 
         costPrice :number;
         sellingPrice :number;
         unit:Unit;
}

export class CompositeProduct extends BaseModel {

        ProductId :number;
        currencyId:number;
        BaseProductId:number; 
        BaseProductUnitId :number;
        Quantity :number;
        TotalCost :number;
        TotalPrice :number;
        BaseProduct:Product;
        Product:Product; 
        UnitProductEquivalence: UnitProductEquivalence;
        Currency: Currency; 
}